### ISSUE TITLE
We need a webgl.poisondata option similar to canvas.poisondata

### ISSUE THREAD

**Issue by ilikenwf** [MoonchildProductions/Pale-Moon#446]
I realize we can disable WebGL, but doing so apparently sends all zeroes. For privacy's sake, it would be nice to reduce our identifying surface size.

Options to entirely disable sending many of the details used on Panopticlick would be nice, actually.

While I use poisondata for the canvas, I'm not sure if it is better or worse to have a random unique ID each time, versus spoofing as some other "known" fingerprint to blend in. 

The secret agent addon is useful for doing this with the browser user agent (blending in). I suppose the jury is out there and that discussion is best left for the forums.

**Comment by wolfbeast** [comment]
You'll have to be a little bit more specific -- which data actually do you ask to be poisoned here?

**Comment by ilikenwf** [comment]
If you go and run a panopticlick test, one of the attributes is "Hash of
WebGL fingerprint" - ours remains the same between page loads. While I
realize that one may not be possible to tamper with easily, simply being
able to disable it and enable it only if a site requires it could be an
option too?

**Comment by wolfbeast** [comment]
Can you please analyze how they create the hash, so I have a bit more to go on? It may possibly not be an option to avoid this since WebGL rendering is off-loaded to the 3D hardware rendering back-end (DirectX/OpenGL).

Only enabling it if a site requires it is not a solution, because a fingerprinting site obviously requires it and there's no way for the browser to distinguish legitimate use from fingerprinting use.

**Comment by ilikenwf** [comment]
I will do some digging, it may very well be the case.

Worst case I may find a way of sending a null instead of zeroes, unless
that's just how panopticlick is displaying it. Because it uses so much js,
I thought they used "undefined" though.

**Comment by ilikenwf** [comment]
It appears that it depends on the browser from some of the fingerprinting
scripts I have looked at. We already have an option to disable WebGL, which
I have opted for personally.

**Comment by wolfbeast** [comment]
Marking this NI then.

**Comment by wolfbeast** [comment]
I've revisited this and as far as I can tell the fingerprinting methods for WebGL rely on drawing on a canvas (pretty much the only way to use WebGL in a page), then getting the data from the canvas with `toDataURL()` which is already covered by the canvas poisondata routine.

The hash not changing between page loads is possibly a bug in the fingerprinting routine, e.g. https://github.com/Valve/fingerprintjs2/pull/20/commits/1ad9a40c339dbf3f648fce39885578f19266ba9d

Closing this as WFM.
